### PR TITLE
Add parameters to initialize Anthropic models with Vertex AI credentials

### PR DESCRIPTION
**Pull request by ShashankVSG**
# Description

Added additional parameters to initialize Anthropic models with Vertex AI credentials.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/base.py [hunk 1]
```diff
@@ -101,6 +102,15 @@ class Anthropic(FunctionCallingLLM):
     max_retries: int = Field(
         default=10, description="The maximum number of API retries.", gte=0
     )
+    vertexai_project_id: Optional[str] = Field(
+        default=None, description="Project ID for VertexAI in Google Cloud"
+    )
+    vertexai_region: Optional[str] = Field(
+        default=None, description="Region for VertexAI in Google Cloud"
+    )
+    vertexai_service_account_file: Optional[str] = Field(
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/base.py:111]: We have a VertexAI LLM class already, why do we need to add this here?

Comment (ShashankVSG) [reply llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/base.py:111]: We can't access anthropic models using that

Comment (logan-markewich) [reply llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/base.py:111]: Then shouldn't the vertexai class be updated no? Or its actually impossible?

Comment (ShashankVSG) [reply llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/base.py:111]: To access anthropic models using vertexai we have to do it through anthropic SDK, which is already used in the llama-index anthropic class.

**Pull request closed without merging**